index.mjs

```diff
@@ -23,28 +27,51 @@ let attempt = 0 // contatore dei tentativi
 let maxGameAttempt = 6; // abbiamo 6 tentativi per risolvere una parola
 let wordGameLength = 4; // la parola deve esser lunga al max 4
 
+
 // Soluzione statica a fine didattico.
-const solution = 'CODE';
+const solution = 'SARA';
 
 // funzione di verifica
 const wordChecker = function (word, solution) {
     let success = [];
     let result = [];
 
+
     // ATTENZIONE ALLE PAROLE maiuscole/minuscole.
+    word = word.toLowerCase();
+    solution = solution.toLowerCase();
 
     // controllo lettera per lettera in parola
     for (let index in word) {
         // VERDE: la lettera è contenuta nella parola nella posizione giusta
         // GIALLO: la lettera è contenuta nella parola ma NON nella posizione giusta
         // GRIGIO: la lettera non è contenuta nella parola
+
+        let letter = word.charAt(index)
+        let trueLetter = solution.charAt(index)
+        let count=0;
+        for(let j=0;j<wordGameLength;j++)
+            if(letter.includes(solution[j]) && j!=index)
+                count++
+        if(letter===trueLetter){
+            result.push(chalk.green(letter.toUpperCase()))
+            success.push(true);
+        }
+        else if(count!=0)
+            result.push(chalk.yellow(letter.toUpperCase()))
+        else    
+            result.push(chalk.grey(letter.toUpperCase()))
+        
     }
+    //Con questo console.log funziona, ma non capisco bene perchè col return non va.
+    //Non capisco bene perchè ciò succeda
+    console.log(result.join(' '))
```

Comment:
Visto! l'esercizio è corretto ma ci sono cose che non abbiamo ancora visto in classe. Pubblico la soluzione cosi la puoi leggere e domani la vediamo tutti insieme in classe!